pages/batch-qrcode.vue

```diff
@@ -0,0 +1,225 @@
+<template>
+  <main class="space-y-4 print:space-y-0">
+    <h1 class="text-lg font-bold font-mono print:hidden">
+      Batch Create QR Codes
+    </h1>
+
+    <UFormGroup
+      label="Upload CSV file"
+      class="print:hidden"
+    >
+      <UInput type="file" accept="csv" @change="handleFileChange" />
+    </UFormGroup>
+    <UDivider class="print:hidden" label="OR" />
+    <UFormGroup
+      class="print:hidden"
+      label="Input CSV content"
+    >
+      <UTextarea
+        v-model="csvInput"
+        class="font-mono"
+        :placeholder="csvInputPlaceholder"
+        :resize="true"
+      />
+    </UFormGroup>
+
+    <div class="print:hidden flex justify-center">
+      <UButton
+        label="QR Code Appearance"
+        variant="outline"
+        @click="isEditingQRCodeConfig = true"
+      />
+    </div>
+
+    <UModal v-model="isEditingQRCodeConfig">
+      <QRCodeGenerator
+        v-model:icon="selectedQRCodeIcon"
+        v-model:color="selectedQRCodeColor"
+        data="https:/books.liker.land"
+        :width="500"
+        :height="500"
+        mode="config"
+        @save="isEditingQRCodeConfig = false"
+      >
+        <template #header>
+          <h3 class="font-bold font-mono">
+            QR Code Config
+          </h3>
+          <UButton
+            icon="i-heroicons-x-mark"
+            color="gray"
+            variant="ghost"
+            @click="isEditingQRCodeConfig = false"
+          />
+        </template>
+      </QRCodeGenerator>
+    </UModal>
+
+    <nav class="flex justify-center items-center gap-2 print:hidden">
+      <UButton
+        v-if="urlItems.length <= 0"
```

Comment:
```suggestion
        v-if="!urlItems?.length"
```
Can't imagine checking <0 would be useful, checking `urlItems.length` exists seem more robust